app/controllers/wishlists_controller.rb

```diff
@@ -21,6 +21,13 @@ def create
     end
   end
 
+  def show
+    @wishlist = Wishlist.find(params[:id])
+    authorize! :show, @wishlist
+  rescue ActiveRecord::RecordNotFound
+    redirect_to root_url, notice: t(".denied")
```

Comment:
And use a more generic error like `failure: "There was an error loading the wishlist."`